File: app/src/main/java/com/example/botcoin/za/co/botcoin/objs/Trade.java
package com.example.botcoin.za.co.botcoin.objs;

public class Trade
{
    public static final String BUY_TYPE = "BUY";
    public static final String SELL_TYPE = "SELL";

    private String type;
    private String amount;
    private String price;


    public Trade(String type, String amount, String price)
    {
        this.type = type;
        this.amount = amount;
        this.price = price;

    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getAmount() {
        return amount;
    }

    public void setAmount(String amount) {
        this.amount = amount;
    }

    public String getPrice() {
        return price;
    }

    public void setPrice(String price) {
        this.price = price;
    }
}

File: app/src/main/java/za/co/botcoin/model/models/Trade.kt
package za.co.botcoin.model.models

import androidx.room.Entity
import androidx.room.Index
import com.google.gson.annotations.SerializedName

@Entity(indices = [Index(value = ["orderId"], unique = true)])
class Trade : BaseModel {
    var type: String = ""
    var volume: String = ""
    var price: String = ""
    var timestamp: String = ""
    var sequence: Int = 0
    var pair: String = ""
    var counter: String = ""
    var base: String = ""

    @SerializedName("client_order_id")
    var clientOrderId: String = ""

    @SerializedName("fee_base")
    var feeBase: String = ""

    @SerializedName("fee_counter")
    var feeCounter: String = ""

    @SerializedName("is_buy")
    var isBuy: Boolean = false

    @SerializedName("order_id")
    var orderId: String = ""

    var accountId: Int = 1

    companion object {
        const val BUY_TYPE = "BUY"
        const val SELL_TYPE = "SELL"
    }

    constructor(type: String, amount: String, price: String) {
        this.type = type
        this.volume = amount
        this.price = price
    }
}